subprojects/code-quality/src/test/groovy/org/gradle/api/plugins/quality/CheckstylePluginMultiProjectTest.groovy

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins.quality
+
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+import org.gradle.util.TestUtil
+
+class CheckstylePluginMultiProjectTest extends AbstractProjectBuilderSpec {
```

Comment:
I'd prefer a test that extends `IntegrationSpec` instead. ProjectBuilder has its quirks and in some corner cases doesn't behave like a real-work `Project` instance.

Reply:
It would make the test awkward. I would need to execute the CheckStyle task and by default it fails with `ModuleVersionNotFoundException: Cannot resolve external dependency com.puppycrawl.tools:checkstyle:6.19 because no repositories are defined.` I think I would be testing too much if I set up Maven repositories and execute a real CheckStyle task just to check that the correct directory is chosen. This test was meant to only test 1 thing: that the path is correct.

Reply:
I am pretty confident that we do not want to add more tests based on `ProjectBuilder`. We'll likely even remove the API at some point. I'd like to ask you turn the test into an `IntegrationSpec` test. You can simple add the functionality to our existing test suite. You'll find plenty of examples under `code-quality/src/integTest`. The current setup also tests the functionality with a range of Checkstyle versions.

Reply:
@pkubowicz Are you still interested in moving forward with this PR?